src/planscape/forsys/views.py

```diff
@@ -3,43 +3,321 @@
 import os
 import pandas as pd
 import rpy2
+import sys
 
+from boundary.models import BoundaryDetails
+from conditions.models import BaseCondition, Condition, ConditionRaster
 from django.conf import settings
+from django.contrib.gis.gdal import CoordTransform, SpatialReference, GDALRaster
+from django.contrib.gis.geos import GEOSGeometry, Point, Polygon
 from django.http import (HttpRequest, HttpResponse, HttpResponseBadRequest,
                          JsonResponse, QueryDict)
 
+
 # Configure global logging.
 logger = logging.getLogger(__name__)
 
-# Gets query parameters for scenario_set
-# TODO: clarify input parameters.
-def get_target_value(params: QueryDict):
-  assert isinstance(params['target_value'], str)
-  target_value = int(params['target_value'])
-  return target_value
-
 # Converts R dataframe to Pandas dataframe.
 # TODO: the broadly-accepted solution involves robjects.conversion.rpy2py - debug why it failed with an input type error.
 def convert_rdf_to_pddf(rdf):
   pddf = pd.DataFrame.from_dict({ key : np.asarray(rdf.rx2(key)) for key in rdf.names })
   return pddf
 
+# Converts dictionary of lists to R dataframe.
+# The lists must have equal length.
+def convert_dictionary_of_lists_to_rdf(lists):
+  data = {}
+  for key in lists.keys():
+    if len(lists[key]) == 0:
+      continue
+    el = lists[key][0]
+    if isinstance(el, str):
+      data[key] = rpy2.robjects.StrVector(lists[key])
+    elif isinstance(el, float):
+      data[key] = rpy2.robjects.FloatVector(lists[key])
+    elif isinstance(el, int):
+      data[key] = rpy2.robjects.IntVector(lists[key])
+
+  rdf = rpy2.robjects.vectors.DataFrame(data)
+  return rdf
+
+# Translates polygon.
```

Comment:
translates from what to what? Looks to me like the input is a polygon and the output also is a polygon, but offset by some amount. Why is this needed?

Reply:
This seems like a language thing: in grade school geometry classes, we learned about translations, rotations, flips, stretches. In this setting, a translation is a "slide" or "offset" - we move an object around a coordinate system without distorting it.

The offset is a temporary hack for making sure an intersection exists between a huc-12 boundary box and a condition raster. I noticed some offset (perhaps not exactly 80 degrees) between the two sources --

sample condition raster bounding boxes
![image](https://user-images.githubusercontent.com/3720361/210422897-65657e12-c168-44e1-9b57-6937ca8890fb.png)

sample huc-12 boundary extents:
![image](https://user-images.githubusercontent.com/3720361/210423150-8d95228e-5d0b-4f8d-998c-9dbf1786567c.png)

I'd strongly prefer to check in code with the translation since that'll give us something that can be demo'd until the actual source of the offset is identified and fixed.

Reply:
SG, thanks for explaining